nck/helpers/facebook_helper.py

```diff
@@ -15,160 +15,125 @@
+
+ACTION_BREAKDOWNS = [v for k, v in AdsInsights.ActionBreakdowns.__dict__.items() if not k.startswith("__")]
+
+def get_action_breakdown_filters(field_path):
+    """
+    Extracts action breakdown filters from a field path,
+    and returns them as a dictionnary.
+    
+    For instance:
+        'actions[action_type:video_view][action_type:post_engagement][action_device:iphone]'
+    returns:
+        {'action_type':['video_view','post_engagement'],
+        'action_device':['iphone']}
+    """
+    filters = {}
+    for path_item in field_path:
+        if ":" in path_item:
+            action_breakdown = path_item.split(":")[0]
+            action_breakdown_value = path_item.split(":")[1]
+            if action_breakdown not in filters:
+                filters[action_breakdown] = [action_breakdown_value]
+            else:
+                filters[action_breakdown].append(action_breakdown_value) 
+    return filters
+
+def format_field_path(field_path):
+    """
+    Formats a field_path back into a field.
+    
+    For instance:
+        ['actions', 'action_type:post_engagement']
+    returns:
+        'actions[action_type:post_engagement]'
+    """
+    if len(field_path)==1:
+        return field_path[0]
+    else:
+        return "".join([field_path[0]] + ["[{}]".format(element) for element in field_path[1:]])
+
+def check_if_obj_meets_action_breakdown_filters(obj, filters):
+    """
+    Checks if a nested action breakdown object
+    meets the conditions defined by action breakdown filters.
+    
+    For instance, if action breakdown filters are:
+        {'action_type': ['post_engagement', 'video_view']
+        'action_device': ['iphone']}
+    Outputs will be:
+    - {'action_type':'post_engagement', 'action_device':'iphone', 'value':12}: True
+    - {'action_type':'video_view', 'action_device':'iphone', 'value':12}: True
+    - {'action_type':'post_engagement', 'action_device':'desktop', 'value':12}: False
+    """
+    obj_meets_all_filters = True
+    for action_breakdown in filters:
+        obj_meets_filter = obj[action_breakdown] in filters[action_breakdown]
+        obj_meets_all_filters = obj_meets_all_filters and obj_meets_filter
+        if obj_meets_all_filters is False:
+            break
+    return obj_meets_all_filters
+
+def get_action_breakdown_value(obj, visited, action_breakdowns):
+    """
+    Extracts the action breakdown value
+    of a nested action breakdown object.
+    
+    For instance:
+        {actions: [{'action_type':'video_view', 'action_device':'iphone', 'value':'12'}]}
+        Here, the nested action_breakdown object is:
+            {'action_type':'video_view', 'action_device':'iphone', 'value':'12'}
+    returns:
+        {'actions[action_type:video_view][action_device:iphone]': '12'}
+    """
+    obj_action_breakdown = ["{}:{}".format(action_breakdown,obj[action_breakdown]) for action_breakdown in action_breakdowns if action_breakdown in obj]
```

Comment:
Ici tu peux utiliser des [f-strings](https://realpython.com/python-f-strings/).

```suggestion
    obj_action_breakdown = [f"{action_breakdown}:{obj[action_breakdown]}" for action_breakdown in action_breakdowns if action_breakdown in obj]
```

C'est plus clair je trouve 😄

Reply:
Yes tu as raison ! Je viens de remplacer tous mes .format( ) par des f-strings.